iroha_actor/src/broker.rs

```diff
@@ -0,0 +1,130 @@
+//! Module with message broker for `iroha_actor`
+
+#![allow(clippy::module_name_repetitions)]
+
+use std::any::{Any, TypeId};
+
+use dashmap::mapref::entry::Entry;
+use dashmap::DashMap;
+use futures::future;
+use once_cell::sync::Lazy;
+
+use super::*;
+
+type TypeMap<V> = DashMap<TypeId, V>;
+type BrokerRecipient = Box<dyn Any + Sync + Send + 'static>;
+
+static BROKER: Lazy<TypeMap<Vec<(TypeId, BrokerRecipient)>>> = Lazy::new(DashMap::new);
+
+/// Trait alias for messages which can be broked
+pub trait BrokerMessage: Message<Result = ()> + Clone + 'static + Send {}
+
+impl<M: Message<Result = ()> + Clone + 'static + Send> BrokerMessage for M {}
+
+fn message_entry<'a>(id: TypeId) -> Entry<'a, TypeId, Vec<(TypeId, BrokerRecipient)>> {
+    BROKER.entry(id)
+}
+
+/// Send message via broker
+pub async fn issue_send<M: BrokerMessage + Send + Sync>(m: M) {
+    let entry = if let Entry::Occupied(entry) = message_entry(TypeId::of::<M>()) {
+        entry
+    } else {
+        return;
+    };
+    let send = entry.get().iter().filter_map(|(_, recipient)| {
+        recipient
+            .downcast_ref::<Recipient<M>>()
+            .map(|recipient| recipient.send(m.clone()))
+    });
+
+    drop(future::join_all(send).await);
+}
+
+/// Trait for using broker
+/// ```rust
+/// use iroha_actor::{prelude::*, broker::{self, *}};
+///
+/// #[derive(Clone)]
+/// struct Message1(String);
+/// impl Message for Message1 { type Result = (); }
+///
+/// #[derive(Clone)]
+/// struct Message2(String);
+/// impl Message for Message2 { type Result = (); }
+///
+/// struct Actor1;
+/// struct Actor2;
+///
+/// #[async_trait::async_trait]
+/// impl Actor for Actor1 {
+///     async fn started(&mut self, ctx: &mut Context<Self>) {
```

Comment:
I think for methods name should be something like `on_start`.

Reply:
This naming is form actix framework. Will change it